src/NetMPA.Catalog/NetMPA.Catalog.Api/Controllers/CatalogController.cs

```diff
@@ -0,0 +1,216 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using NetMPA.Catalog.Api.Controllers.Requests;
+using NetMPA.Catalog.Api.Models;
+using NetMPA.Catalog.Bll.Interfaces.Services;
+//using NetMPA.Catalog.Bll.Models;
+
+namespace NetMPA.Catalog.Api.Controllers
+{
+    /// <summary>
+    /// Catalog API
+    /// </summary>
+    [ApiVersion("1.0")]
+    [Route("api/v{version:apiVersion}/[controller]")]
+ 
+    [ApiController]
+    public class CatalogController : ControllerBase
+    {
+        private readonly ILogger<CatalogController> logger;
+        private readonly ICategoryService categoryService;
+        private readonly IProductService productService;
+        private readonly IMapper mapper;
+
+        public CatalogController(ICategoryService categoryService, IProductService productService,IMapper mapper, ILogger<CatalogController> logger)
+        {
+            this.logger = logger;
+            this.categoryService = categoryService;
+            this.productService = productService;
+            this.mapper = mapper;
+        }
+
+        /// <summary>
+        /// List of Categories
+        /// </summary>
+        [HttpGet()]
+        [Route("category")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<IEnumerable<Category>>> GetAllCategories([FromQuery] PagedRequest pagedRequest)
+        {
+            var result = await this.categoryService.List();
+            if (result == null) return NotFound();
+            return Ok(result.Take(pagedRequest.Take).Skip(pagedRequest.Skip).Select(this.mapper.Map<Category>));
+        }
+
+        /// <summary>
+        /// Category by Id
+        /// </summary>
+        [HttpGet()]
+        [Route("category/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<Category>> GetCategoryById(int id)
+        {
+            var result = await this.categoryService.Get(id);
+            if (result == null) return NotFound();
+            return Ok(this.mapper.Map<Category>(result));
+        }
+
+        /// <summary>
+        /// Add Category
+        /// </summary>
+        [HttpPost]
+        [Route("category")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult> AddCategory([FromBody] CategoryAdd category)
+        {
+            if (category == null) return BadRequest();
+
+            await this.categoryService.Add(new Bll.Models.Category()
+                {
+                    Name = category.Name,
+                    Image = category.Image,
+                    Parent = new Bll.Models.Category() { Id = category.ParentId }
+                }
+            );
+
+            return StatusCode(201);
```

Comment:
Standard way in ASP.NET Core for this is to use [CreatedAtAction](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.createdataction?view=aspnetcore-7.0)

Reply:
surem but for this one I have to have an Id as result from service. So have ommit it for now